src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,45 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    public void getStatistic(String fromFile, String toFile) {
+        int supplyTotal = 0;
+        int buyTotal = 0;
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFile))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(",");
+                if (parts.length != 2) {
+                    continue;
+                }
+                String operation = parts[0].trim();
+                int amount = Integer.parseInt(parts[1].trim());
```

Parsing the amount uses `Integer.parseInt(parts[1].trim())` but parsing errors (for example a malformed integer) are not wrapped with a contextual RuntimeException. The checklist requires handling malformed lines and not ignoring exceptions: "Handle potential malformed lines gracefully according to chosen error policy; at minimum, ensure parsing uses split(",") and that amount is parsed as an integer, throwing a RuntimeException with context if parsing fails (derivable from exception handling requirement) [CHECKLIST ITEM #12]." Catch `NumberFormatException` around parsing (or validate before parsing) and throw a `RuntimeException` that includes the input file and the offending line. See parsing at this line .